### PR TITLE
feat: implement API for downloading and reading reports

### PR DESCRIPTION
**Pull request by shadowusr**
### What's done?

Implemented API that allows to download and read reports programmatically. Currently marked as experimental.

Currently used in Testplane MCP/CLI integrations with HTML Reporter.

### CODE REVIEW

File: lib/sdk.ts [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+}
+
+const DEFAULT_REPORT_FILES_TO_DOWNLOAD: ReportFileToDownload[] = ['dbFiles'];
+
+const isString = (value: unknown): value is string => typeof value === 'string';
+
+const resolveAttempt = (attemptsByBrowser: Map<string, number>, row: RawSuitesRow): number => {
+    const testPath: string[] = JSON.parse(row[DB_COLUMN_INDEXES.suitePath] as string);
+    const browserName = row[DB_COLUMN_INDEXES.name] as string;
+    const browserId = [...testPath, browserName].join(' ');
```
Comment (sipayRT) [inline lib/sdk.ts:39]: но это же не browserId

File: test/unit/lib/db-utils/common.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+            let err;
+            try {
+                await promise;
+            } catch (e) {
+                err = e;
+            }
+
+            assert.include(err.message, 'Cannot download file from "/report/databaseUrls.json"');
+            assert.include(err.message, 'request failed with status 200');
```
Comment (sipayRT) [inline test/unit/lib/db-utils/common.js:54]: неоднозначное сообщение, в котором рядом `failed` и `status 200`. лучше явно написать, что в ответе не получили данных

**Review by sipayRT — APPROVED** [review]

File: test/unit/lib/db-utils/common.js (hunk 2 above)
Comment (shadowusr) [reply test/unit/lib/db-utils/common.js:54]: Это худший edge-case, где у нас нет сообщения об ошибке, а есть только статус. Тогда мы пишем такое сообщение. В 90% кейсов там красивое сообщение с конкретной причиной падения.

File: lib/sdk.ts (hunk 1 above)
Comment (shadowusr) [reply lib/sdk.ts:39]: В HTML Reporter мы повсеместно используем browserId именно в таком значении, например, при работе со стейтом и т.д. Так исторически сложилось (еще до меня) и в целом если привыкнуть, то логично. Под browserId понимается id теста, который заканчивается браузером.
